src/web/js/output-ui.js

```diff
@@ -685,278 +443,234 @@
         }
       });
     }
-      
-    function highlightSrcloc(runtime, editors, srcloc, loc, cssColor, context, underline) {
-      if (underline === undefined) underline = true;
-      var styles = document.getElementById("highlight-styles").sheet;
-      return runtime.ffi.cases(runtime.getField(srcloc, "is-Srcloc"), "Srcloc", loc, {
-        "builtin": function(_) { /* no-op */ },
-        "srcloc": function(source, startL, startC, startCh, endL, endC, endCh) {
-          var cmLoc = cmPosFromSrcloc(runtime, srcloc, loc);
-          var locKey = cmlocToCSSClass(cmLoc);
-          var editor = editors[source];
-          if(editor) {
-            styles.insertRule(
-                  ((context === undefined) ? "" : "#main[data-highlights=" + context + "]")
-                   + " ." + locKey + " { " + (!!cssColor ? "background-color:" + cssColor : "") +
-                  (underline ? ";border-bottom: 2px hsla(0, 0%, 0%,.5) solid" : "") + ";}",
-                  styles.cssRules.length);
-            addMark(source, editor, cmLoc.start, cmLoc.end,
-                    {className: locKey + " highlight", shared: false});
-          }
-          return undefined;
-        }
-      });
-    }
 
-    function highlightLines(runtime, editors, srcloc, loc, cssColor, context) {
-      var styles = document.getElementById("highlight-styles").sheet;
-      return runtime.ffi.cases(runtime.getField(srcloc, "is-Srcloc"), "Srcloc", loc, {
-        "builtin": function(_) { /* no-op */ },
-        "srcloc": function(source, startL, startC, startCh, endL, endC, endCh) {
-          var cmLoc = cmPosFromSrcloc(runtime, srcloc, loc);
-          var locKey = cmlocToCSSClass(cmLoc);
-          var editor = editors[source];
-          if(editor) {
-            styles.insertRule(
-                  ((context === undefined) ? "" : "#main[data-highlights=" + context + "]")
-                  + " ." + locKey + " { background-color:" + cssColor + ";}",
-                  0);
-            for(var i=cmLoc.start.line;i<=cmLoc.end.line;i++){
-              editor.addLineClass(i,"background",locKey);
-            }
-            return locKey;
+    var Snippet = function () {
```

Comment:
Yeah, I dislike the pattern, but I'm not going to argue strongly against if it's used consistently.